torchvision/prototype/transforms/_transform.py

```diff
@@ -54,6 +58,56 @@ def extra_repr(self) -> str:
 
         return ", ".join(extra)
 
+    # This attribute should be set on all transforms that have a v1 equivalent. Doing so enables the v2 transformation
+    # to be scriptable. See `_extract_params_for_v1_transform()` and `__prepare_scriptable__` for details.
+    _v1_transform_cls: Optional[Type[nn.Module]] = None
+
+    def _extract_params_for_v1_transform(self) -> Dict[str, Any]:
+        # This method is called by `__prepare_scriptable__` to instantiate the equivalent v1 transform from the current
+        # v2 transform instance. It does two things:
+        # 1. Extract all available public attributes that are specific to that transform and not `nn.Module` in general
+        # 2. If available handle the `fill` attribute for v1 compatibility (see below for details)
+        # Overwrite this method on the v2 transform class if the above is not sufficient. For example, this might happen
+        # if the v2 transform introduced new parameters that are not support by the v1 transform.
+        common_attrs = nn.Module().__dict__.keys()
+        params = {
+            attr: value
+            for attr, value in self.__dict__.items()
+            if not attr.startswith("_") and attr not in common_attrs
+        }
+
+        if "fill" not in params:
+            return params
+
+        # transforms v2 has a more complex handling for the `fill` parameter than v1. By default, the input is parsed
```

Comment:
How much more clunky would it be to move the `fill` handling in the child classes that actually support it? It is in general not desirable to have a special-case handling of subclasses in the base-class

Reply:
We have 6 transforms that need it:

- `Pad`
- `ElasticTransform`
- `RandomAffine`
- `RandomCrop`
- `RandomPerspective`
- `RandomRotation`

Given that we shouldn't need to touch this after release since v1 will no longer be changed, I'm ok with keeping the special case here.

In general, I agree with you though. If you would like to have the handling in the subclasses, I'm not going to oppose.

Reply:
OK, let's just leave it there then

Reply:
I'll add the transforms above to the comment so no one has to go digging if we decide otherwise in the future.

Reply:
OK but keep it short :) there's already a lot of comments